### DISCUSSION TITLE
BLtouch ABL mesh not correct

### DISCUSSION THREAD

**Discussion by Srizzyy** [Jyers/Marlin#688] [Q&A] [answered]
Dear Jyers , many thanks first of all for your efforts, i’m having an issue with your firmware about the leveling , when i set the z offset propertly and then ask to get the mesh leveling done and both getting saved from control setting, then i start the print of the squares around the corners , i don’t get the same resutls at all on the 9 squares, now with the new leveling menu you did i had to tune the mesh manually and found a huge gap in every point . But eventually after all the tuning all squares printed well, but why is this issue happening, why can’t the automatic mesh leveling do the job ? I’m afraid that i have to. Do the tuning everytime i remove the bed.

The firmware used is the BLtouch hs 5x5 v4.2.2 

Till now haven’t seen any other issues :D

**Comment by Jyers** [comment, marked as answer]
Nothing has changed about the way auto leveling is handled so my best guess as to the cause of this would either be a hardware fault of some kind, or an error caused by not clearing eeprom after flashing the firmware.

**Reply by Srizzyy** [reply]
This issue is happening with previous versions as well, but now when i can see tuning , i decided to give it a try and it helped alot!, i’m not assuming fully that it is a software issue, what is clearing eeprom after flashing a firmware ? And is eeprom saving must be with sd card inserted ? 

Thanks

**Reply by Jyers** [reply]
No, the SD doesn't need to be in to save to eeprom. To clear eeprom, go to the control menu, select reset to defaults and then store settings. Of course make sure you write down any settings you want to keep before doing this.
